network/src/peer/mod.rs

```diff
@@ -54,9 +55,9 @@ pub enum PeerNotification {
     PeerDisconnected(Identity, Multiaddr, ConnectionOrigin, DisconnectReason),
 }
 
-pub struct Peer<TMuxer>
+pub struct Peer<TSocket>
 where
-    TMuxer: StreamMultiplexer,
+    TSocket: AsyncRead + AsyncWrite + Send + Debug + Unpin + Sync + 'static,
```

Comment:
It might be useful to write a composite trait: https://github.com/libra/libra/blob/master/consensus/consensus-types/src/common.rs#L16